lib/associations/belongs-to-many.js

```diff
@@ -367,66 +367,49 @@ BelongsToMany.prototype.injectGetter = function(obj) {
 };
 
 BelongsToMany.prototype.injectSetter = function(obj) {
-  var association = this
-    , primaryKeyAttribute = association.target.primaryKeyAttribute;
+  var association = this;
 
   obj[this.accessors.set] = function(newAssociatedObjects, options) {
     options = options || {};
-    var instance = this;
+    var instance = this
+      , sourceKey = association.source.primaryKeyAttribute
+      , targetKey = association.target.primaryKeyAttribute
+      , identifier = association.identifier
+      , foreignIdentifier = association.foreignIdentifier
+      , where = {};
+
+    if (newAssociatedObjects === null) {
+      newAssociatedObjects = [];
+    } else {
+      newAssociatedObjects = association.toInstanceArray(newAssociatedObjects);
+    }
```

Comment:
`newAssociatedObjects = !newAssociatedObjects && [] || association.toInstanceArraay(newAssociatedObjects)`
Not sure if that's less clear though.